[pytorch] [distributed] [easy] Added test case for reinit

satgera opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26506 [pytorch] [distributed] [easy] Added test case for reinit**

[pytorch] [distributed] Made test forgiving to allow rpc agent to return one of the two errors.

Differential Revision: [D17488899](https://our.internmc.facebook.com/intern/diff/D17488899/)

satgera committed:
[pytorch] [distributed] [easy] Made test forgiving to allow rpc agent to return one of the two errors

[pytorch] [distributed] Made test forgiving to allow rpc agent to return one of the two errors.

Differential Revision: [D17488899](https://our.internmc.facebook.com/intern/diff/D17488899/)

[ghstack-poisoned]

pritamdamania87 commented on test/test_rpc.py:
```diff
@@ -133,7 +133,12 @@ def test_reinit(self):
         dist.init_process_group(backend="gloo", rank=self.rank,
                                 world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel(self_name='worker{}'.format(self.rank),
```
Shouldn't we have two separate tests here? One uses duplicate worker name and raises an error with message "is not unique" and then another initializes model_parallel again with a unique name but should raise an error with "is already initialized"?

satgera committed:
Update on "[pytorch] [distributed] [easy] Made test forgiving to allow rpc agent to return one of the two errors"


[pytorch] [distributed] Made test forgiving to allow rpc agent to return one of the two errors.

Differential Revision: [D17488899](https://our.internmc.facebook.com/intern/diff/D17488899/)

[ghstack-poisoned]

satgera commented:
@pytorchbot retest this please

pietern commented on test/test_rpc.py:
```diff
@@ -129,17 +129,42 @@ def test_self_add(self):
             dist.init_model_parallel(self_name="duplicate_name",
                                      backend=BACKEND,
                                      self_rank=self.rank,
                                      init_method=RPC_INIT_URL)
```
AFAICT this shouldn't initialize with `init_model_parallel` outside the `assertRaisesRegex` wrapper.

The purpose is to only test for duplicate names right? Not that it was already initialized.

satgera committed:
Update on "[pytorch] [distributed] [easy] Added test case for reinit"


[pytorch] [distributed] Made test forgiving to allow rpc agent to return one of the two errors.

Differential Revision: [D17488899](https://our.internmc.facebook.com/intern/diff/D17488899/)

[ghstack-poisoned]

pietern reviewed (approved):
LGTM, provided CI passes.

satgera commented:
@pytorchbot retest this please

satgera commented:
Retried the CI to ensure all pass but looks like infra issue. Some of the failure which indicate build failure seems to have succeed.
Since this is a trivial change and other test have passed, landing.
